modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/DmlPlannerTest.java

```diff
@@ -0,0 +1,121 @@
+     */
+    @Test
+    public void testInsertIntoSingleDistributedTable() throws Exception {
+        IgniteTable test1 = createTable("TEST1", IgniteDistributions.single(), "C1", Integer.class, "C2", Integer.class);
+        IgniteSchema schema = createSchema(test1);
+
+        // There should be no exchanges and other operations.
+        assertPlan("INSERT INTO TEST1 (C1, C2) VALUES(1, 2)", schema,
+                isInstanceOf(IgniteTableModify.class).and(input(isInstanceOf(IgniteValues.class))));
+    }
+
+    /**
+     * Test for INSERT .. VALUES when table has non single distribution.
+     */
+    @ParameterizedTest
+    @MethodSource("nonSingleDistributions")
+    public void testInsert(IgniteDistribution distribution) throws Exception {
+        IgniteTable test1 = createTable("TEST1", distribution, "C1", Integer.class, "C2", Integer.class);
+
+        IgniteSchema schema = createSchema(test1);
+
+        assertPlan("INSERT INTO TEST1 (C1, C2) VALUES(1, 2)", schema,
+                nodeOrAnyChild(isInstanceOf(IgniteExchange.class)
+                        .and(e -> e.distribution().equals(IgniteDistributions.single())))
+                        .and(nodeOrAnyChild(isInstanceOf(IgniteTableModify.class))
+                                .and(hasChildThat(isInstanceOf(IgniteExchange.class).and(e -> distribution.equals(e.distribution())))))
+        );
+    }
+
+    private static Stream<IgniteDistribution> nonSingleDistributions() {
+        return distributions().filter(d -> !IgniteDistributions.single().equals(d));
+    }
+
+    /**
+     * Test for INSERT .. FROM SELECT when tables has different distributions.
+     */
+    @ParameterizedTest
+    @MethodSource("distributions")
+    public void testInsertSelectFrom(IgniteDistribution distribution) throws Exception {
+        IgniteDistribution anotherDistribution = IgniteDistributions.affinity(1, new UUID(1, 0), "0");
+
+        IgniteTable test1 = createTable("TEST1", distribution, "C1", Integer.class, "C2", Integer.class);
+        IgniteTable test2 = createTable("TEST2", anotherDistribution,  "C1", Integer.class, "C2", Integer.class);
+
+        IgniteSchema schema = createSchema(test1, test2);
+
+        assertPlan("INSERT INTO TEST1 (C1, C2) SELECT C1, C2 FROM TEST2", schema,
+                nodeOrAnyChild(isInstanceOf(IgniteExchange.class)
+                        .and(e -> e.distribution().equals(IgniteDistributions.single())))
+                        .and(nodeOrAnyChild(isInstanceOf(IgniteTableModify.class))
+                                .and(hasChildThat(isInstanceOf(IgniteExchange.class).and(e -> distribution.equals(e.distribution())))))
+        );
+    }
+
+    /**
+     * Test for INSERT .. FROM SELECT when tables has the same distribution.
+     */
+    @ParameterizedTest
+    @MethodSource("distributions")
+    public void testInsertSelectFromSameDistribution(IgniteDistribution distribution) throws Exception {
```

Comment:
I would like to add one more test for `UPDATE table SET a = a + 1` query

Reply:
Added a test case for that.